duckduckgo/Android
# Generate webview image when leaving BrowserActivity

Task/Issue URL: https://app.asana.com/0/1126768040926570/1140520481588209
Tech Design URL: 
CC: 

**Description**:
This fixes an issue whereby a page might be loaded in a tab (TAB A) but the callback for page loaded is fired before anything is visible, producing a blank WebView preview. This would normally be fine if the user hit the tab switcher button. But if they instead leave the app, then they might return to the app later launching another tab (TAB B). From TAB B, they could launch the tab switcher. TAB B preview would be re-generated, but TAB A would remain white.

This PR ensures the preview is generated when the user is leaving the screen altogether, and this removes the need to generate a bitmap every time the page loads.


**Steps to test this PR**:
1. Launch a tab, and browse to a site; **do not** hit the tab switcher
1. Hit the home button
1. Launch the app into a new tab (e.g, use the widget, long press the app for the new tab shortcut etc...)
1. Hit the tab switcher button
1. Verify the original tab preview is populated
1. Generally browse around, open new tabs, and verify the webview preview logic holds


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
